src/mediaelement-webaudio.js

```diff
@@ -59,6 +59,13 @@ export default class MediaElementWebAudio extends MediaElement {
         this.sourceMediaElement.connect(this.analyser);
     }
 
+    play(start, end) {
+        if (this.ac.state == 'suspended') {
+            this.ac.resume && this.ac.resume();
```

Comment:
Hi, because `resume` method of AudioContext return a `Promise`, maybe is better to call `play` when Promise is solved? @thijstriemstra what do you thing?

Reply:
Makes sense.

Reply:
Hi folks, 
I tried this approach but Safari is throwing me:

```
Unhandled Promise Rejection: NotAllowedError: The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.
```

(this appears to get thrown from the actual call to `play`).  I think it's because when we try to call `super.play()` from the `ac.resume` promise, we leave the context of the `click` handler and Safari thinks we're trying to auto-play audio.

So I think the code is probably correct as-is.

Reply:
> we leave the context of the click handler and Safari thinks we're trying to auto-play audio.

Did you try `resume`-ing in the error handler of the promise? I guess the audiocontext/state is destroyed/polluted by then and `resume` doesn't make a difference anymore.

Reply:
also, this check/statement is also present in webaudio.js, perhaps we can move this check to some utility function and re-use it?

Reply:
> Did you try resume-ing in the error handler of the promise? I expect the audiocontext/state is destroyed/polluted by then and resume doesn't make a difference anymore. 

I was trying like:

```
  ac.resume().then(() => super.play(...)
```

which seems like the most natural code to write.  The call to `resume` succeeds, but the call here: 

https://github.com/katspaugh/wavesurfer.js/blob/b7d1a25965869acb1843323232831ad48f5780af/src/mediaelement.js#L291

throws `NotAllowedError`.  If you have another approach you'd like me to try let me know.